### PR TITLE
Task Entity and Migration

### PR DESCRIPTION
**Pull request by akib-raihan-sazim**
**Description of changes**
- Define Task entity using MikroORM.
- Setup db migration for the Task entity.

**Associated ticket link**
https://trello.com/c/LizHUMZA

**Screenshot / Screen Recording**
N/A

### CODE REVIEW

File: src/common/entities/task.entity.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  Medium = "Medium",
+  Low = "Low",
+}
+
+@Entity()
+export class Task {
+  @PrimaryKey()
+  id!: number;
+
+  @Property()
```
Comment (ragib-hossain-sazim) [inline src/common/entities/task.entity.ts:14]: isn't title a mandatory field?

File: src/common/entities/task.entity.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  title!: string;
+
+  @Property()
+  summary!: string;
+
+  @Property({ type: "date", nullable: true })
```
Comment (ishmam-iqbal-sazim) [inline src/common/entities/task.entity.ts:20]:
```suggestion
  @Property({ type: "date", nullable: true, fieldName: "due_date" })
```

Please be explicit about the fieldName in your entity file. All Property decorator inside entity file should contain fieldName for that attribute.

File: src/common/entities/task.entity.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { Entity, PrimaryKey, Property, Enum } from "@mikro-orm/core";
+
+export enum EPriority {
+  High = "High",
+  Medium = "Medium",
+  Low = "Low",
+}
```
Comment (ishmam-iqbal-sazim) [inline src/common/entities/task.entity.ts:7]: Create a folder inside common called enums and create a file as per the entity name. In your case task.enums.ts and put the relevant enums inside that.

```
common
  |
  |__entities
  |    |__task.entity.ts
  |
  |__enums
       |__task.enums.ts
```

File: src/common/entities/task.entity.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { Entity, PrimaryKey, Property, Enum } from "@mikro-orm/core";
+
+export enum EPriority {
```
Comment (ishmam-iqbal-sazim) [inline src/common/entities/task.entity.ts:3]:
```suggestion
export enum ETaskPriority {
```

Let's try to be as precise as possible with the naming

**Review by ishmam-iqbal-sazim — APPROVED** [review]

File: .gitignore [hunk 5]
```diff
@@ -54,3 +54,8 @@ pids
 
 # Diagnostic reports (https://nodejs.org/api/report.html)
 report.[0-9]*.[0-9]*.[0-9]*.[0-9]*.json
+
+# Temp file
+temp/
+temp/Task.js.json
```
Comment (ragib-hossain-sazim) [inline .gitignore:60]: /temp should exclude everything from the folder, no need to specify the file name

File: src/db/migrations/.snapshot-to-do.json
Comment (ragib-hossain-sazim) [inline src/db/migrations/.snapshot-to-do.json:1]: .snapshot.json files shouldn't be included as part of commit, please add them in .gitIgnore.

**Review by ragib-hossain-sazim — APPROVED** [review]
Approved pending comments

**Pull request merged**